<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorWidget</class>
 <widget class="QWidget" name="PasswordGeneratorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>575</width>
    <height>284</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>284</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <item>
    <layout class="QGridLayout" name="passwordFieldLayout">
     <property name="bottomMargin">
      <number>10</number>
     </property>
     <property name="verticalSpacing">
      <number>0</number>
     </property>
     <item row="1" column="1">
      <widget class="QProgressBar" name="entropyProgressBar">
       <property name="minimumSize">
        <size>
         <width>50</width>
         <height>5</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>5</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QProgressBar {
	border: none;
	height: 2px;
    font-size: 1px;
	background-color: transparent;
    padding: 0 1px;
}
QProgressBar::chunk {
	background-color: #c0392b;
	border-radius: 2px;
}</string>
       </property>
       <property name="maximum">
        <number>200</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="textVisible">
        <bool>false</bool>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
       <property name="textDirection">
        <enum>QProgressBar::TopToBottom</enum>
       </property>
       <property name="format">
        <string>%p%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="labelNewPassword">
       <property name="text">
        <string>Password:</string>
       </property>
       <property name="buddy">
        <cstring>editNewPassword</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="passwordStrengthTextLayout">
       <item>
        <widget class="QLabel" name="strengthLabel">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="text">
          <string>strength</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="entropyLabel">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>entropy</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="1">
      <widget class="PasswordEdit" name="editNewPassword">
       <property name="maxLength">
        <number>999</number>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="labelLength">
       <property name="text">
        <string>&amp;Length:</string>
       </property>
       <property name="buddy">
        <cstring>spinBoxLength</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1" colspan="2">
      <layout class="QHBoxLayout" name="passwordLengthSliderLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QSlider" name="sliderLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>128</number>
         </property>
         <property name="sliderPosition">
          <number>20</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item alignment="Qt::AlignRight">
        <widget class="QSpinBox" name="spinBoxLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="2">
      <widget class="QToolButton" name="togglePasswordButton">
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="optionsLayout">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Character Types</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="alphabetLayout" stretch="0,0,0,0,1">
          <item>
           <widget class="QToolButton" name="checkBoxUpper">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="toolTip">
             <string>Upper Case Letters</string>
            </property>
            <property name="text">
             <string notr="true">A-Z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxLower">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="toolTip">
             <string>Lower Case Letters</string>
            </property>
            <property name="text">
             <string notr="true">a-z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxNumbers">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="toolTip">
             <string>Numbers</string>
            </property>
            <property name="text">
             <string notr="true">0-9</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxSpecialChars">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="toolTip">
             <string>Special Characters</string>
            </property>
            <property name="text">
             <string notr="true">/*_&amp; ...</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxExcludeAlike">
          <property name="text">
           <string>Exclude look-alike characters</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">optionButtons</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxEnsureEvery">
          <property name="text">
           <string>Pick characters from every group</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">optionButtons</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="globalButtonsLayout">
       <item>
        <widget class="QPushButton" name="buttonGenerate">
         <property name="text">
          <string>Generate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonApply">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Accept</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PasswordEdit</class>
   <extends>QLineEdit</extends>
   <header>gui/PasswordEdit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editNewPassword</tabstop>
  <tabstop>togglePasswordButton</tabstop>
  <tabstop>sliderLength</tabstop>
  <tabstop>spinBoxLength</tabstop>
  <tabstop>checkBoxUpper</tabstop>
  <tabstop>checkBoxLower</tabstop>
  <tabstop>checkBoxNumbers</tabstop>
  <tabstop>checkBoxSpecialChars</tabstop>
  <tabstop>checkBoxExcludeAlike</tabstop>
  <tabstop>checkBoxEnsureEvery</tabstop>
  <tabstop>buttonGenerate</tabstop>
  <tabstop>buttonApply</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="optionButtons">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
